bitcoin/src/psbt/map/global.rs

```diff
@@ -51,10 +51,13 @@ impl Map for PartiallySignedTransaction {
         }
 
         // Serializing version only for non-default value; otherwise test vectors fail
-        if self.version > 0 {
+        if self.version > Version::PsbtV0 {
```

Comment:
I'm not sure exactly why but my gut tells me we should match on the variants explicitly and remove `PartialOrd`/`Ord`. For the "keep the highest version" (in `combine`) we could use `to_raw` then compare the values.